frontend/lib/src/components/widgets/NumberInput/NumberInput.tsx

```diff
@@ -263,38 +268,35 @@ export const NumberInput: React.FC<Props> = ({
     ]
   )
 
-  const onBlur = (): void => {
+  const onBlur = useCallback((): void => {
     if (dirty) {
       commitValue({ value, source: { fromUi: true } })
     }
     setIsFocused(false)
-  }
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [dirty])
 
-  const onFocus = (): void => {
+  const onFocus = useCallback((): void => {
     setIsFocused(true)
-  }
+  }, [])
 
-  const updateFromProtobuf = (): void => {
+  const updateFromProtobuf = useCallback((): void => {
     const { value } = element
     element.setValue = false
     setValue(value ?? null)
     setFormattedValue(formatValue({ value: value ?? null, ...element, step }))
     commitValue({ value: value ?? null, source: { fromUi: false } })
-  }
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [element, step])
```

Comment:
since we use `useCallback` for `commitValue`, it should be a stable reference when nothing changes and so we should be able to pass it to the deps array and get rid of the eslint-disablement. Maybe I am missing something though.